File: java/app/src/main/java/com/youknow/movie/ui/nowplaying/NowPlayingContract.java
package com.youknow.movie.ui.nowplaying;

import com.youknow.domain.model.SimpleMovie;

import java.util.List;

public interface NowPlayingContract {
    interface View {
        void showProgressBar(int visibility);
        void onMoviesLoaded(List<SimpleMovie> movies);
        void onError(int msgResId);
        void hideError();
    }

    interface Presenter {
        void getUpcomingMovies();
    }
}

File: kotlin-rx-koin/app/src/main/java/com/youknow/movie/ui/nowplaying/NowPlayingContract.kt
package com.youknow.movie.ui.nowplaying

import com.youknow.domain.model.SimpleMovie

interface NowPlayingContract {
    interface View {
        fun showProgressBar(visibility: Int)
        fun onMoviesLoaded(movies: List<SimpleMovie>)
        fun onError(msgResId: Int)
        fun hideError()

    }

    interface Presenter {
        fun unsubscribe()
        fun getMoviesNowPlaying()

    }
}